app/controllers/UkimsNumberController.scala

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import controllers.actions._
+import forms.UkimsNumberFormProvider
+import play.api.i18n.I18nSupport
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
+import views.html.UkimsNumberView
+
+import javax.inject.Inject
+
+class UkimsNumberController @Inject() (
+  val controllerComponents: MessagesControllerComponents,
+  authorise: AuthoriseAction,
+  view: UkimsNumberView,
+  formProvider: UkimsNumberFormProvider
+) extends FrontendBaseController
+    with I18nSupport {
+  private val form = formProvider()
+
+  def onPageLoad: Action[AnyContent] = authorise { implicit request =>
+    Ok(view(form))
+  }
+
+  def onSubmit: Action[AnyContent] = authorise { implicit request =>
+    //TODO saving session data???
```

Comment:
can you also add to fix the redirect to the TODO comment

Reply:
This is done